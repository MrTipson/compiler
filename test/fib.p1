void printf(char* format, void* arg1, void* arg2);
void putint(int val);

int fibonacci(int n){
	if(n < 2){
		return n;
	}
	int a = 0;
	int b = 1;
	while(n-- >= 2){
		int c = a + b;
		a = b;
		b = c;
	}
	return b;
}

void main(int argc, char** argv) {
	int x = 10;
	printf("Fibonacci(%d)=%d", (void*)x, (void*)fibonacci(x));
}