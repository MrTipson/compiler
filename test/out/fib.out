Fibonacci(10)=55